pypesto/petab/importer.py

```diff
@@ -415,6 +436,37 @@ def create_objective(
             par_ids = [x for x in par_ids if x not in inner_parameter_ids]
 
             # FIXME: currently not supported with hierarchical
+            if 'guess_steadystate' in kwargs and kwargs['guess_steadystate']:
+                warnings.warn(
+                    "`guess_steadystate` not supported with hierarchical optimization. Disabling `guess_steadystate`."
+                )
+            kwargs['guess_steadystate'] = False
+
+        if self._ordinal:
+            inner_solver_options = kwargs.pop('inner_solver_options', None)
+            inner_solver_options = (
+                inner_solver_options
+                if inner_solver_options is not None
+                else self._inner_solver_options
+            )
+            inner_problem_method = inner_solver_options.get('method', None)
+
+            inner_problem = OptimalScalingProblem.from_petab_amici(
+                self.petab_problem, model, edatas, inner_problem_method
+            )
+            inner_solver = OptimalScalingInnerSolver(
+                options=inner_solver_options
+            )
+            calculator = OptimalScalingAmiciCalculator(
+                inner_problem, inner_solver
+            )
+            amici_reporting = amici.RDataReporting.full
```

Comment:
Good point.
So we do need from amici ReturnData the simulations and sensitivities, so I need to set the reporting to something more than just likelihood, but asking Daniel now, it turns out I don't need `...full` and that I can just use the `...residuals` one as it has all of that too and is faster.

Reply:
It seems like we can do this with the hierarchical one as well as it needs only simulations and sigma's from ReturnData. @yannikschaelte should I change it from `amici.RDataReporting.full` to `amici.RDataReporting.residuals` there too?

Reply:
`RDataReporting.residuals` will not give us the gradient. Might be a good idea to think about some extra mode as suggested by @FFroehlich, but that's out of scope here.

Reply:
gradient is accessible as `res.dot(sres)` though.

Reply:
Usually yes, for ordinal measurements it's not `res.dot(sres)`. There's surrogate data instead of measurements, which depends on the outer parameters. So optimal scaling still needs `sy`.